pkg/copy/copy.go

```diff
@@ -0,0 +1,83 @@
+package copy
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/crane"
+	"github.com/google/go-containerregistry/pkg/name"
+)
+
+// CopyImage copies a single container image with its cosign signature from source
+// to target registry. The source and target must be fully qualified image references
+// including registry, repository, and tag.
+//
+// Example:
+//
+//	err := copy.CopyImage(
+//	    "stgregistry.suse.com/rancher/fleet-agent:v0.13.0",
+//	    "localhost:5000/rancher/fleet-agent:v0.13.0",
+//	)
+func CopyImage(sourceImage, targetImage string) error {
```

Comment:
Part of this logic already exists [here](https://github.com/rancherlabs/slsactl/blob/main/internal/imagelist/copier.go#L106). Ideally we would extract the common parts and place it in a location that the existing code can call.